From 098851ffb6c5855e74fd5f80300fc9c2758e2677 Mon Sep 17 00:00:00 2001
From: Trent Mick <trentm@gmail.com>
Date: Thu, 25 Jul 2019 10:22:00 -0700
Subject: [PATCH] TOOLS-2298 check-copyright fails for non-master Jenkins
 builds Reviewed by: Tim Foster <tim.foster@joyent.com> Approved by: Tim
 Foster <tim.foster@joyent.com>

---
 tools/check-copyright | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/tools/check-copyright b/tools/check-copyright
index b260b93..9fa6130 100755
--- a/tools/check-copyright
+++ b/tools/check-copyright
@@ -145,7 +145,13 @@ done
 shift $((OPTIND - 1))
 
 # Gather the files to possibly check.
-committedFiles=$(git diff --name-only master..)
+committedFiles=
+if git branch -a | grep '^  remotes/origin/master$' >/dev/null 2>/dev/null; then
+    # The intent for committed files is to compare against the ultimate
+    # merge branch target. We assume that is "origin/master", the common case
+    # for Joyent eng repos.
+    committedFiles=$(git diff --name-only origin/master..)
+fi
 stagedFiles=$(git diff --name-only --staged)
 unstagedFiles=$(git diff --name-only)
 uniqFiles=$(echo "
-- 
2.21.0

